### PR TITLE
React hook form

### PR DESCRIPTION
**Pull request by marianozicavo**
## Summary

- Replace formik with react-hook-form 🎉 🎉 🎉 
- With formik, we had a HOC to help us create the formik fields. The RHF equivalent is the [controller](https://react-hook-form.com/api#Controller). I thought a nice approach to this would be separating the controlled component (such as `CustomTextInput`) from the controller. So I made a `controller.tsx` file next to the `index.tsx`. My idea would be to have a `controller.tsx` for each of the controlled components we have in the project, so we can define custom `defaultValue`s, `onChange` functions, etc.
In some projects we used the `withForm` HOC for a lot of different controlled components and it had a lot of nasty logic inside so it could be used with every component. I think having a completely separate controller for each component is better.
- Added `ref` functionality to `CustomTextInput`

## Screenshots
image
image
image
image

## Trello Card

https://trello.com/c/bx9PGa1F/195-migrar-manejo-de-formularios-de-formik-a-react-hook-form

### CODE REVIEW

File: generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js [hunk 1]
```diff
@@ -40,15 +40,9 @@ describe('<Login />', () => {
...
-    });
-    await waitFor(async () => {
-      await fireEvent.changeText(passwordInput, VALID_PASSWORD);
-    });
-    await waitFor(async () => {
-      await fireEvent.press(submitButton);
-    });
+    fireEvent.changeText(emailInput, VALID_EMAIL);
+    fireEvent.changeText(passwordInput, VALID_PASSWORD);
+    await waitFor(() => fireEvent.press(submitButton));
```
Comment (mcavo) [inline generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js:45]: Did you check if you don't put a valid email and password that submit is not been called? `expect(dispatch).toHaveBeenCalledTimes(0);`

**Review by mcavo — COMMENTED** [review]
Great job!!! Just one question 😄

File: generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js (hunk 1 above)
Comment (marianozicavo) [reply generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js:50]: Good idea! I added a new expect

**Review by mcavo — APPROVED** [review]

File: generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js [hunk 2]
```diff
@@ -9,14 +9,18 @@ const VALID_PASSWORD = 'HelloWord1234';
...
     const { getByText, getAllByText, getByTestId } = render(<Login />);
 
     const submitButton = getByText('Iniciar sesión');
     const emailInput = getByTestId('Email');
 
     fireEvent.press(submitButton);
-    await waitFor(() => getAllByText('Este campo es obligatorio'));
-    expect(getAllByText('Este campo es obligatorio')).toHaveLength(2);
+
+    await waitFor(() => expect(getAllByText('Este campo es obligatorio')).toHaveLength(2));
```
Comment (guidoprinc) [inline generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js:23]: shouldn't this be a i18n response?

Comment (marianozicavo) [reply generators/app/templates/src/app/screens/Auth/screens/Login/index.test.js:23]: I agree that we should use i18n in the tests. But I'd tackle that in another PR, when we add additional tests to the bootstrap

**Review by mattgle — APPROVED** [review]
NO MORE FORMIK! HELL YEAH!

**Review by guidoprinc — APPROVED** [review]

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**